apps/logger/logger.go

```diff
@@ -8,17 +6,32 @@ import (
 	"log/slog"
 )
 
+// CallbackFunc defines the signature for callback functions
+// we can only have one string to support azure sdk
+type CallbackFunc func(level, message string)
+
+type Level string
+
+const (
+	Info  Level = "info"
+	Err   Level = "error"
+	Warn  Level = "warn"
+	Debug Level = "debug"
+)
+
 // Logger struct for Go 1.21+ with full `slog` logging support.
 type Logger struct {
-	logging     *slog.Logger
-	logCallback CallbackFunc
+	logging *slog.Logger
 }
 
-func New121(input interface{}) (*Logger, error) {
-	if logger, ok := input.(*slog.Logger); ok {
-		return &Logger{logging: logger}, nil
+// New creates a new logger instance
+func New(slogLogger *slog.Logger) (*Logger, error) {
```

Comment:
I am not sure about this. The moment we put slog in the public API surface, Azure SDK will not be able to use MSAL SDK.

Reply:
See with https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/main/apps/design/release.md for testing with azure sdk